contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -1000,10 +953,80 @@ contract RibbonTreasuryVault is
     }
 
     /**
-     * @notice Burn the remaining oTokens left over from gnosis auction.
+     * @notice Settles the round's Gnosis auction and distribute the premiums earned
+     */
+    function concludeOptionsSale() external onlyKeeper nonReentrant {
+        VaultLifecycleTreasury.settleAuction(
+            GNOSIS_EASY_AUCTION,
+            optionAuctionID
+        );
+
+        _chargeAndDistribute();
+    }
+
+    /**
+     * @notice Charge performance fee and distribute remaining to whitelisted address
+     */
+    function chargeAndDistribute() external onlyKeeper nonReentrant {
+        require(
+            IERC20(USDC).balanceOf(address(this)) > 0,
+            "no premium to distribute"
+        );
```

Comment:
Shouldn't this check go into `_chargeAndDistribute`?

Reply:
This was to prevent double checks. in `commitAndClose` we are already checking balance before calling `_chargeAndDistribute`.

Reply:
Added this guard to `_chargeAndDistribute`. For functions where `_chargeAndDistribute` is optional (e.g. `concludeOptionsSale` and `commitAndClose`, the function will check balance and only call `_chargeAndDistribute` when balance is larger than 0.